pilot/pkg/networking/core/listener_inbound.go

```diff
@@ -225,21 +236,36 @@ func (lb *ListenerBuilder) buildInboundListeners() []*listener.Listener {
 		// to handle mTLS vs plaintext and HTTP vs TCP (depending on protocol and PeerAuthentication).
 		var opts []FilterChainMatchOptions
 		mtls := lb.authnBuilder.ForPort(cc.port.TargetPort)
-		// Chain has explicit user TLS config. This can only apply when the TLS mode is DISABLE to avoid conflicts.
-		if cc.tlsSettings != nil && mtls.Mode == model.MTLSDisable {
+
+		var chains []*listener.FilterChain
+		// Chain has explicit user TLS config. This can only apply when the TLS mode is DISABLE or PERMISSIVE to ensure
+		// security.
+		if cc.tlsSettings != nil && mtls.Mode != model.MTLSStrict {
+			// copy as may be used to build istio mTLS listener
+			cc := cc
+			mtls := mtls
+
 			// Since we are terminating TLS, we need to treat the protocol as if its terminated.
 			// Example: user specifies protocol=HTTPS and user TLS, we will use HTTP
 			cc.port.Protocol = cc.port.Protocol.AfterTLSTermination()
+
+			// In permissive mode, both the Istio mTLS filter chains and user TLS filter chains
+			// are built. Set the flag to ensure when constructing them they are uniquely referencable.
+			if mtls.Mode == model.MTLSPermissive {
+				cc.permissive = true
+			}
 			lp := istionetworking.ModelProtocolToListenerProtocol(cc.port.Protocol)
 			opts = getTLSFilterChainMatchOptions(lp)
 			mtls.TCP = BuildListenerTLSContext(cc.tlsSettings, lb.node, lb.push.Mesh, istionetworking.TransportProtocolTCP, false)
 			mtls.HTTP = mtls.TCP
-		} else {
+			chains = append(chains, lb.inboundChainForOpts(cc, mtls, opts)...)
+		}
+
+		if cc.tlsSettings == nil || mtls.Mode != model.MTLSDisable {
 			lp := istionetworking.ModelProtocolToListenerProtocol(cc.port.Protocol)
```

Comment:
TODO: Will need to ensure that the permissive TLS chains are not built when `tlsSettings != nil`.

Reply:
Addressed in [8d29fdd](https://github.com/istio/istio/pull/53469/commits/8d29fdd1aaa3b6f1150ab32555306b693db6ec23)